api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/PersonAttributeTranslatorImpl.java

```diff
@@ -0,0 +1,198 @@
+	
+	private static final String OPENMRS_CONCEPT_FORMAT = "org.openmrs.Concept";
+	
+	private static final String LOCATION_TYPE = "Location";
+	
+	@Autowired
+	private LocationService locationService;
+	
+	@Autowired
+	private PersonService personService;
+	
+	@Autowired
+	private ConceptService conceptService;
+	
+	@Autowired
+	private ConceptTranslator conceptTranslator;
+	
+	@Override
+	public Extension toFhirResource(@Nonnull PersonAttribute personAttribute) {
+		if (personAttribute.getAttributeType() == null) {
+			throw new IllegalArgumentException("PersonAttribute and AttributeType cannot be null");
+		}
+		
+		Extension wrapperExtension = createWrapperExtension();
+		Extension valueExtension = createValueExtension(personAttribute);
+		wrapperExtension.setExtension(Collections.singletonList(valueExtension));
+		
+		return valueExtension;
+	}
+	
+	@Override
+	public PersonAttribute toOpenmrsType(@Nonnull Extension extension) {
+		if (!isValidPatientAttributeExtension(extension)) {
+			return null;
+		}
+		
+		PersonAttribute personAttribute = new PersonAttribute();
+		Extension valueExtension = extension.getExtensionFirstRep();
+		Type extensionValue = valueExtension.getValue();
+		String attributeTypeName = extractAttributeTypeName(valueExtension.getUrl());
+		
+		personAttribute.setAttributeType(getPersonAttributeType(attributeTypeName));
+		setPersonAttributeValue(personAttribute, extensionValue);
+		
+		return personAttribute;
+	}
+	
+	private Extension createWrapperExtension() {
+		return new Extension(new UriType(FhirConstants.OPENMRS_FHIR_EXT_PERSON_ATTRIBUTE));
+	}
+	
+	private Extension createValueExtension(PersonAttribute personAttribute) {
+		PersonAttributeType attributeType = personAttribute.getAttributeType();
+		String extensionUrl = FhirConstants.OPENMRS_FHIR_EXT_PERSON_ATTRIBUTE_TYPE + "#" + attributeType.getName();
+		Extension extension = new Extension(new UriType(extensionUrl));
+		
+		setExtensionValue(extension, personAttribute);
+		return extension;
+	}
+	
+	private void setExtensionValue(Extension extension, PersonAttribute personAttribute) {
+		String format = personAttribute.getAttributeType().getFormat();
+		String value = personAttribute.getValue();
+		
+		try {
+			switch (format) {
+				case JAVA_STRING_FORMAT:
+					extension.setValue(new StringType(value));
+					break;
+				case JAVA_BOOLEAN_FORMAT:
+					extension.setValue(new BooleanType(value));
+					break;
+				case OPENMRS_LOCATION_FORMAT:
+					handleLocationFormat(extension, value);
+					break;
+				case OPENMRS_CONCEPT_FORMAT:
+					handleConceptFormat(extension, value);
+					break;
+				default:
+					throw new UnsupportedOperationException("Unsupported attribute format: " + format);
+			}
+		}
+		catch (Exception e) {
+			throw new IllegalStateException("Error processing attribute value for format: " + format, e);
+		}
+	}
+	
+	private void handleLocationFormat(Extension extension, String locationId) {
```

Comment:
This seems to rely on exposing the primary key identifier for the location. These values must _always_ be converted to and from UUIDs when sent out.